hack/ci-e2e-kind-migration.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o pipefail
+set -o errexit
+
+if [[ "$OSTYPE" != "darwin"* ]]; then
+# https://github.com/kubernetes/test-infra/issues/23741
+iptables -t mangle -A POSTROUTING -p tcp --tcp-flags SYN,RST SYN -j TCPMSS --clamp-mss-to-pmtu
+fi
+
+# test setup
+make kind-up
+export KUBECONFIG=$PWD/gardener-local/kind/kubeconfig
+make gardener-up
+
+# setup second kind cluster
+make kind2-up
+make gardenlet-kind2-up
+
```

Comment:
Now that we have two kind clusters, I assume https://github.com/gardener/gardener/blob/bdd5621e4e882be510d08ae01bc1f7774803efbb/hack/test-e2e-local.sh#L17-L19 will only dump the logs of one of them, right? Can this be improved?

Reply:
Just one question here: From what I understand in @timebertt's proposal in https://github.com/gardener/gardener/pull/6089#issuecomment-1164029426 the suggestion is to also move the `dump_logs()` trap inside `ci-e2e-kind.sh`. Should we then also add it to `ci-e2e-kind-migration.sh`, but have it dump the logs for the second kind cluster? One thing that I see as an issue here is that `kind` inside scripts called from Makefile automatically refers to `tools/bin/kind`, but this would not be the case for `ci-e2e-kind-migration.sh`

Reply:
> Should we then also add it to `ci-e2e-kind-migration.sh`, but have it dump the logs for the second kind cluster?

Yes, that would be the plan.



> One thing that I see as an issue here is that `kind` inside scripts called from Makefile automatically refers to `tools/bin/kind`, but this would not be the case for `ci-e2e-kind-migration.sh`

Good catch!
That means we would probably need make targets `ci-e2e-kind*` for that. That would be the easiest option.

Reply:
> That means we would probably need make targets ci-e2e-kind* for that. That would be the easiest option.
That would be my idea as well. I'll do it like that :)